crates/cli/src/option.rs

```diff
@@ -0,0 +1,177 @@
+use anyhow::{bail, Result};
+use std::path::PathBuf;
+
+/// An option group used for parsing strings to their group option representation.
+#[derive(Clone, Debug)]
+pub struct GroupOption<T>(pub Vec<T>);
+
+#[derive(Clone)]
+pub struct GroupOptionParser<T>(pub std::marker::PhantomData<T>);
+
+/// Generic option attributes.
+#[derive(Debug)]
+pub struct OptionMeta {
+    pub name: String,
+    pub doc: String,
+    pub help: String,
+}
+
+pub fn fmt_help(cmd: &str, short: &str, meta: &[OptionMeta]) {
+    println!("Available options for {}", cmd);
+    for opt in meta {
+        println!();
+        print!("-{:<3}", short);
+        print!("{:>3}", opt.name);
+        println!("{}", opt.help);
+        for line in opt.doc.split('\n') {
+            print!("{}", line);
+        }
+        println!();
+    }
+}
+
+/// Commonalities between all option groups.
+// Until we make more extensive use of this trait.
+#[allow(dead_code)]
+pub trait GroupDescriptor {
+    fn options() -> Vec<OptionMeta>;
+}
+
+pub trait GroupOptionBuilder: Clone + Sized + Send + Sync + 'static {
+    fn parse(val: &str) -> anyhow::Result<Self>
+    where
+        Self: Sized;
+}
+
+pub fn to_kebab_case(val: &str) -> String {
+    let kebab_case = val
+        .chars()
+        .flat_map(|c| {
+            if c.is_uppercase() {
+                vec!['-', c.to_ascii_lowercase()]
+            } else {
+                vec![c]
+            }
+        })
+        .collect::<String>();
+
+    let name = if let Some(stripped) = kebab_case.strip_prefix('-') {
+        stripped
+    } else {
+        &kebab_case
+    };
+
+    name.to_string()
+}
+
+#[macro_export]
+macro_rules! option_group {
```

Comment:
I mean... this should really be a Derive macro, shouldn’t it? But totally understand if you can’t be bothered to create a subcrate etc etc

Reply:
Yeah, I really didn't want to create another crate and go through that dance. Perhaps a thing to do in a follow up.